src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -73,458 +72,409 @@ const defaultProps = {
-            return CONST.IOU.AMOUNT_MAX_LENGTH + 1;
+    useEffect(() => {
+        if (isEditing.current) {
+            if (prevMoneyRequestID.current !== props.iou.id) {
+                // The ID is cleared on completing a request. In that case, we will do nothing.
+                if (props.iou.id) {
+                    Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), true);
+                }
+                return;
+            }
+            const moneyRequestID = `${iouType.current}${reportID.current}`;
+            const shouldReset = props.iou.id !== moneyRequestID;
+            if (shouldReset) {
+                IOU.resetMoneyRequestInfo(moneyRequestID);
+            }
+
+            if (_.isEmpty(props.iou.participants) || props.iou.amount === 0 || shouldReset) {
+                Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), true);
+            }
         }
 
-        /*
-        Return the sum of leading zeroes length and absolute amount length(including fraction digits).
-        When the absolute amount is 0, add 2 to the leading zeroes length to represent fraction digits.
-        */
-        return leadingZeroesLength + (absAmount === '0' ? 2 : absAmount.length);
-    }
+        return () => {
+            prevMoneyRequestID.current = props.iou.id;
+        };
+    }, [props.iou.participants, props.iou.amount, props.iou.id]);
 
-    /**
-     * Check if amount is a decimal up to 3 digits
-     *
-     * @param {String} amount
-     * @returns {Boolean}
-     */
-    validateAmount(amount) {
-        const decimalNumberRegex = new RegExp(/^\d+(,\d+)*(\.\d{0,2})?$/, 'i');
-        return amount === '' || (decimalNumberRegex.test(amount) && this.calculateAmountLength(amount) <= CONST.IOU.AMOUNT_MAX_LENGTH);
-    }
+    useEffect(() => {
+        if (!props.route.params.currency) {
+            return;
+        }
 
-    /**
-     * Strip comma from the amount
-     *
-     * @param {String} amount
-     * @returns {String}
-     */
-    stripCommaFromAmount(amount) {
-        return amount.replace(/,/g, '');
-    }
+        setSelectedCurrencyCode(props.route.params.currency);
+    }, [props.route.params.currency]);
 
-    /**
-     * Strip spaces from the amount
-     *
-     * @param {String} amount
-     * @returns {String}
-     */
-    stripSpacesFromAmount(amount) {
-        return amount.replace(/\s+/g, '');
-    }
+    useEffect(() => {
+        setSelectedCurrencyCode(props.iou.currency);
+    }, [props.iou.currency]);
+
+    useEffect(() => {
+        const selectedAmountAsStringForState = props.iou.amount ? CurrencyUtils.convertToWholeUnit(props.iou.currency, props.iou.amount).toString() : '';
+        setAmount(selectedAmountAsStringForState);
+        setSelection({
+            start: selectedAmountAsStringForState.length,
+            end: selectedAmountAsStringForState.length,
+        });
+    }, [props.iou.amount, props.iou.currency]);
```

Comment:
Hi, this PR caused a regression/bug - [Currency changes after refresh although URL has currency params](https://github.com/Expensify/App/issues/23436#top)
#23436

more details here - https://github.com/Expensify/App/issues/23436#issuecomment-1646938220

Reply:
Interesting! Previously we used the `componentDidUpdate` method which only occurs after an update while the `useEffect` is triggered also on the initial render (ignoring dependencies).